File: app/discover/scan_instance.rb
#!/usr/bin/env ruby

# scan an instance

require 'singleton'
require_relative 'db_fetch_instance_details'
require_relative 'scanner'

class ScanInstance < Scanner
  include Singleton
  
  def initialize()
    super([
      {:fetcher => DbFetchHostInstances.new()}
    ])
  end
  
end

File: app/discover/scan_instance.py
# scan a host for instances
from discover.folder_fetcher import FolderFetcher
from discover.scanner import Scanner
from utils.singleton import Singleton


class ScanInstance(Scanner, metaclass=Singleton):
    def __init__(self):
        super(ScanInstance, self).__init__([
            {
                "type": "vnics_folder",
                "fetcher": FolderFetcher("vnics", "instance", "vNICs"),
                "children_scanner": "ScanVnicsRoot"
            }
        ])
